Source/Layer.js

```diff
@@ -7,317 +7,405 @@
-    */
+  /**
+  Duplicate this layer.
+  A new identical layer will be inserted into the parent of this layer.
 
-    duplicate() {
-      return self._document.wrapObject(this._object.duplicate());
-    }
+  @return {Layer} A new layer identical to this one.
+  */
 
-    /**
-        Is this a page?
+  duplicate() {
+    return self._document.wrapObject(this._object.duplicate());
+  }
 
-        All Layer objects respond to this method, but only pages return true.
+  /**
+  Is this a page?
 
-        @return {bool} true for instances of Group, false for any other layer type.
-    */
+  All Layer objects respond to this method, but only pages return true.
 
-    get isPage() { return false; }
+  @return {bool} true for instances of Group, false for any other layer type.
+  */
 
-    /**
-        Is this an artboard?
+  get isPage() { return false; }
 
-        All Layer objects respond to this method, but only Artboard objects return true.
+  /**
+  Is this an artboard?
 
-        @return true for instances of Artboard, false for any other layer type.
-    */
+  All Layer objects respond to this method, but only Artboard objects return true.
 
-    get isArtboard() { return false; }
+  @return true for instances of Artboard, false for any other layer type.
+  */
 
-    /**
-        Is this an group?
+  get isArtboard() { return false; }
 
-        All Layer objects respond to this method, but only Groups or things that inherit from groups return true.
+  /**
+  Is this an group?
 
-        @return {bool} true for instances of Group, false for any other layer type.
-    */
+  All Layer objects respond to this method, but only Groups or things that inherit from groups return true.
 
-    get isGroup() { return false; }
+  @return {bool} true for instances of Group, false for any other layer type.
+  */
 
-    /**
-        Is this a text layer?
+  get isGroup() { return false; }
 
-        All Layer objects respond to this method, but only text layers return true.
+  /**
+  Is this a text layer?
 
-        @return {bool} true for instances of Group, false for any other layer type.
-    */
+  All Layer objects respond to this method, but only text layers return true.
 
-    get isText() { return false; }
+  @return {bool} true for instances of Group, false for any other layer type.
+  */
 
-    /**
-        Is this a shape layer?
+  get isText() { return false; }
 
-        All Layer objects respond to this method, but only shape layers (rectangles, ovals, paths etc) return true.
+  /**
+  Is this a shape layer?
 
-        @return {bool} true for instances of Group, false for any other layer type.
-    */
+  All Layer objects respond to this method, but only shape layers (rectangles, ovals, paths etc) return true.
 
-    get isShape() { return false; }
+  @return {bool} true for instances of Group, false for any other layer type.
+  */
 
-    /**
-        Is this an image layer?
+  get isShape() { return false; }
```

Comment:
Could do, but this applies to the javascript classes themselves, not to the underlying traits. I'm deliberately simplifying the view in the API, so I didn't necessarily want to expose the traits.